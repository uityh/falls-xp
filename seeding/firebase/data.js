import { db } from './index';
import { collection, addDoc } from '@firebase/firestore';

export const addUser = async (user) => {
	await addDoc(collection(db, 'users'), user);
};

export const addProject = async (project) => {
	try {
		const projectRef = await addDoc(collection(db, 'projects'), {
			first: 'Ada',
			last: 'Lovelace',
			born: 1815,
		});
		console.log('Document written with ID: ', projectRef.id);
	} catch (e) {
		console.error('Error adding document: ', e);
	}
};
