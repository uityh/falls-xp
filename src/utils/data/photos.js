import {
	getStorage,
	ref,
	uploadBytes,
	deleteObject,
	getDownloadURL,
} from 'firebase/storage';

export const uploadFile = (file, filePath) => {
	const storage = getStorage();
	const storageRef = ref(storage, filePath);
	uploadBytes(storageRef, file)
		.then((snapshot) => {
			console.log('Uploaded a blob or file!');
			return getDownloadURL(snapshot.ref);
		})
		.catch((error) => {
			console.log(error);
		});
};

export const deleteFile = (filePath) => {
	const storage = getStorage();
	const storageRef = ref(storage, filePath);
	deleteObject(storageRef)
		.then(() => {
			console.log('File deleted successfully');
		})
		.catch((error) => {
			console.log(error);
		});
};
