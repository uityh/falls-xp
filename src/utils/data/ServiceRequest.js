import { collection, addDoc } from 'firebase/firestore';
import { db } from 'utils/firebase';
import { checkString, checkDateString } from 'utils/helpers/validation';

export const createServiceRequest = async (customerId, salesRepId, address, startDate, customerNotes) => {

	customerId = checkString(customerId, 'customerId');
	salesRepId = checkString(salesRepId, 'salesRepId');
	address = checkString(address, 'address');
	startDate = checkDateString(startDate, 'startDate');
	customerNotes = checkString(customerNotes, 'customerNotes');

	startDate.setHours(0, 0, 0);

	return await addDoc(collection(db, 'projects'), { 
		customerId: customerId,
		salesRepId: salesRepId,
		startDate: startDate,
		address: address,
		onsiteWorkers: [],
		status: "not started",
		tasks: ["initial inspection"],
		cost: 0,
		customerNotes: customerNotes
	});
}