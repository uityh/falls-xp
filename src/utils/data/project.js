import { collection, arrayUnion, updateDoc, getDoc } from 'firebase/firestore';
import { db } from 'utils/firebase';

export const getImageUrls = async (projectId) => {
	const project = await getDoc(doc(db, 'projects', projectId));
	return project.data().imageUrls;
};

export const addImageUrl = async (projectId, imageUrl) => {
	await updateDoc(doc(db, 'projects', projectId), {
		imageUrls: arrayUnion(imageUrl),
	});
	return getImageUrls(projectId);
};
