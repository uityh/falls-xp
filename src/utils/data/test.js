import { collection, getDocs } from 'firebase/firestore';
import { db } from 'utils/firebase';

export const getTestData = async () => {
	const snapshot = await getDocs(collection(db, 'test'));
	const result = [];
	snapshot.forEach((doc) => {
		result.push({
			id: doc.id,
			...doc.data(),
		});
	});
	return result;
};

export const getTestDataById = async (id) => {
	const doc = await getDocs(collection(db, 'test', id));
	return doc.data();
};
