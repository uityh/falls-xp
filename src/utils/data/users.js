import {
	collection,
	getDocs,
	getDoc,
	doc,
	addDoc,
	query,
	where,
} from 'firebase/firestore';
import {
	createUserWithEmailAndPassword,
	sendPasswordResetEmail,
	signInWithEmailAndPassword,
} from 'firebase/auth';
import { auth, db } from 'utils/firebase';

export const getAllUsers = async () => {
	const usersCollection = await getDocs(collection(db, 'users'));
	const result = [];
	usersCollection.forEach((userDoc) => {
		result.push({
			id: userDoc.id,
			...userDoc.data(),
		});
	});
	return result;
};

export const getUserById = async (id) => {
	const userDoc = await getDoc(doc(db, 'users', id));
	return {
		id,
		...userDoc.data(),
	};
};

export const getUserByEmail = async (email) => {
	//!
	console.log('Before getDocs call');
	const userDoc = await getDocs(
		query(collection(db, 'users'), where('email', '==', email))
	);
	const result = [];
	//!
	console.log('Before docs forEach loop');
	userDoc.forEach((uDoc) => {
		result.push({
			id: uDoc.id,
			...uDoc.data(),
		});
	});
	//!
	console.log('after forEach loop');
	console.log(result);
	console.log(userDoc);
	if (result.length === 0) throw new Error('No user with provided email found');
	return result[0];
};

const checkEmailExists = async (email) => {
	try {
		const user = await getUserByEmail(email);
		if (user) return true;
	} catch (e) {
		return false;
	}
	return false;
};

export const createUser = async (user) => {
	if (checkEmailExists(user.email))
		throw new Error('An account with this email already exists');
	const userDoc = await addDoc(collection(db, 'users'), user);
	await createUserWithEmailAndPassword(auth, user.email, userDoc.id);
	await sendPasswordResetEmail(auth, user.email);
	return getUserById(userDoc.id);
};

export const authenticateUser = async (email, password) => {
	try {
		//!
		console.log('Before calling signInWithEmailAndPassword');
		const res = await signInWithEmailAndPassword(auth, email, password);
		//!
		console.log('Successfully calles signIn function');
		const { user } = res;
		//!
		console.log('Before getUserByEmail');
		const userObj = await getUserByEmail(user.email);
		//!
		console.log('After getUserByemail');
		return userObj;
	} catch (e) {
		throw new Error('Email or password is incorrect');
	}
};
