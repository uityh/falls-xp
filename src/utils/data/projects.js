/* eslint-disable no-console */
/* eslint-disable */
import {
	collection,
	getDocs,
	getDoc,
	updateDoc,
	arrayUnion,
	doc,
	query,
	where,
	addDoc,
} from 'firebase/firestore';
import { db } from 'utils/firebase';
import { getUserById } from 'utils/data/users';
import { checkString, checkDateString } from 'utils/helpers/validation';

export const getAllProjects = async (testdb) => {
	if (testdb) {
		const projectsCollection = await testdb().collection('projects').get();
		const result = [];
		projectsCollection.forEach((projectDoc) => {
			result.push({
				id: projectDoc.id,
				...projectDoc.data(),
			});
		});
		return result;
	}
	const projectsCollection = await getDocs(collection(db, 'projects'));
	const result = [];
	projectsCollection.forEach((projectDoc) => {
		result.push({
			id: projectDoc.id,
			...projectDoc.data(),
		});
	});
	return result;
};

export const getProjectByProjectId = async (id) => {
	const projectDoc = await getDoc(doc(db, 'projects', id));
	if (!projectDoc) throw new Error('No project exists for the given Id');
	return {
		id,
		...projectDoc.data(),
	};
};

export const checkProjectInvolvement = async (projectId, userId) => {
	const projectData = await getProjectByProjectId(projectId);
	const thisUser = await getUserById(userId);
	if (thisUser.role === 'admin' || thisUser.role === 'sales') {
		return true;
	}
	if (thisUser.role === 'customer') {
		if (projectData.customerId === userId) {
			return true;
		}
		return false;
	}
	if (projectData.assignedWorkers.includes(userId)) {
		return true;
	}
	return false;
};

export const getProjectByInvolvedId = async (id) => {
	const thisUser = await getUserById(id);
	const result = [];
	let foundProjects = null;
	if (thisUser.role === 'admin') {
		return getAllProjects();
	}
	if (thisUser.role === 'customer') {
		foundProjects = query(
			collection(db, 'projects'),
			where('customerId', '==', id)
		);
	} else if (thisUser.role === 'sales') {
		foundProjects = query(
			collection(db, 'projects'),
			where('salesRepId', '==', id)
		);
	} else {
		foundProjects = query(
			collection(db, 'projects'),
			where('assignedWorkers', 'array-contains', id)
		);
	}
	const projectsDoc = await getDocs(foundProjects);
	if (!projectsDoc) throw new Error('No projects found for the current user');
	projectsDoc.forEach((projectDoc) => {
		result.push({
			id: projectDoc.id,
			...projectDoc.data(),
		});
	});
	return result;
};

export const getImageUrls = async (projectId, testdb) => {
	if (testdb) {
		const project = await testdb().collection('projects').doc(projectId).get();
		return project.data().imageUrls;
	}
	const project = await getDoc(doc(db, 'projects', projectId));
	return project.data().imageUrls;
};

export const addImageUrl = async (projectId, imageUrl) => {
	await updateDoc(doc(db, 'projects', projectId), {
		imageUrls: arrayUnion(imageUrl),
	});
	return getImageUrls(projectId);
};

export const markTaskAsComplete = async (projectId, taskName, taskNotes) => {
	const foundProject = await getProjectByProjectId(projectId);
	const idx = foundProject.tasks.findIndex((task) => task.name === taskName);
	foundProject.tasks[idx].status = 'complete';
	foundProject.tasks[idx].endDate = today;
	foundProject.tasks[idx].taskNotes = taskNotes;

	await updateDoc(doc(db, 'projects', projectId), {
		tasks: foundProject.tasks,
	});

	return foundProject.tasks;
};

export const addTaskToProject = async (
	projectId,
	taskName,
	taskNotes = '',
	completePreviousTask = false
) => {
	const taskTeamRelations = {
		'initial inspection': 'onsite',
		'site review': 'operations',
		'customer confirmation': 'sales',
		installation: 'onsite',
	};
	taskName = checkString(taskName);
	if (!Object.keys(taskTeamRelations).includes(taskName))
		throw new Error('Invalid task name');
	if (taskNotes !== '') taskNotes = checkString(taskNotes);
	if (typeof completePreviousTask !== 'boolean')
		throw new Error('Invalid completePreviousTask value (must be boolean)');

	const foundProject = await getProjectByProjectId(projectId);
	if (!foundProject) throw new Error('No project found for the given id');
	const today = new Date();
	today.setHours(0, 0, 0, 0);
	const task = {
		name: taskName,
		status: 'in progress',
		startDate: today,
		endDate: endDate,
		team: taskTeamRelations[taskName],
		taskNotes: taskNotes,
	};
	foundProject.tasks.push(task);

	if (foundProject.tasks.length > 1 && completePreviousTask) {
		const idx = foundProject.tasks.findIndex((task) => task.name === taskName);
		foundProject.tasks[idx - 1].status = 'complete';
		foundProject.tasks[idx - 1].endDate = today;
	}

	await updateDoc(doc(db, 'projects', projectId), {
		tasks: foundProject.tasks,
	});

	return foundProject.tasks;
};

export const getProjectsByCustomerId = async (customerId, testdb) => {
	if (testdb) {
		const userDoc = await testdb()
			.collection('users')
			.where('customerId', '==', customerId);
		const result = [];
		userDoc.forEach((uDoc) => {
			result.push({
				id: uDoc.id,
				...uDoc.data(),
			});
		});
		return result;
	}

	const userDoc = await getDocs(
		query(collection(db, 'projects'), where('customerId', '==', customerId))
	);
	const result = [];
	userDoc.forEach((uDoc) => {
		result.push({
			id: uDoc.id,
			...uDoc.data(),
		});
	});
	return result;
};

export const getProjectsByStatus = async (status) => {
	const projectsDoc = await getDocs(
		query(collection(db, 'projects'), where('status', '==', status))
	);
	let result = [];
	projectsDoc.forEach((pDoc) => {
		result.push({
			id: pDoc.id,
			...pDoc.data(),
		});
	});
	result = await Promise.all(
		result.map(async (p) => {
			const customer = await getUserById(p.customerId);
			return {
				...p,
				customer,
			};
		})
	);
	return result;
};

export const getProjectsForSales = async () => {
	const projectsDoc = await getDocs(
		query(collection(db, 'projects'), where('status', '!=', 'lead onboarded'))
	);
	let result = [];
	projectsDoc.forEach((pDoc) => {
		result.push({
			id: pDoc.id,
			...pDoc.data(),
		});
	});
	result = await Promise.all(
		result.map(async (p) => {
			const customer = await getUserById(p.customerId);
			return {
				...p,
				customer,
			};
		})
	);
	return result;
};

export const createNewProject = async (customerId, address = '') => {
	if (!customerId) {
		throw new Error('A customer ID is required to create a new project');
	}

	const thisCustomer = await getUserById(customerId);
	if (thisCustomer.role !== 'customer') {
		console.log('Error from backend');
		throw new Error('That is not a real customer!');
	}

	return addDoc(collection(db, 'projects'), {
		customerId,
		address,
		salesRepId: '',
		startDate: '',
		assignedWorkers: [],
		status: 'lead onboarded',
		tasks: [],
		cost: 0,
		customerNotes: '',
		imageUrls: [],
	});
};

export const getProjectLeads = async () => {
	const projectDoc = await getDocs(
		query(collection(db, 'projects'), where('status', '==', 'lead onboarded'))
	);
	let result = [];
	projectDoc.forEach((pDoc) => {
		result.push({
			id: pDoc.id,
			...pDoc.data(),
		});
	});
	result = await Promise.all(
		result.map(async (p) => {
			const customer = await getUserById(p.customerId);
			return {
				...p,
				customer,
			};
		})
	);
	return result;
};

export const createServiceRequest = async (
	projectIdParam,
	salesRepIdParam,
	startDateParam,
	customerNotesParam
) => {
	const projectId = checkString(projectIdParam, 'projectId');
	await getProjectByProjectId(projectId);

	const salesRepId = checkString(salesRepIdParam, 'salesRepId');
	const startDate = checkDateString(startDateParam, 'startDate');
	let customerNotes = '';
	if (customerNotesParam !== '') {
		customerNotes = checkString(customerNotesParam, 'customerNotes');
	}

	const task = {
		taskName: 'initial inspection',
		startDate,
		status: 'in progress',
		team: 'onsite',
		taskNotes: '',
	};

	await updateDoc(doc(db, 'projects', projectId), {
		status: 'initial inspection',
		salesRepId,
		customerNotes,
		startDate,
		tasks: [task],
	});

	return getProjectByProjectId(projectId);
};

// Create service request is the functionality to be used for creating a project
