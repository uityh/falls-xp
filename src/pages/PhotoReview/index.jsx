/* eslint-disable */

import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';

import {
	Box,
	Typography,
	Card,
	Button,
	IconButton,
	FormControl,
	TextField,
	Select,
	MenuItem
} from '@mui/material';

import ArrowLeftRoundedIcon from '@mui/icons-material/ArrowLeftRounded';
import ArrowRightRoundedIcon from '@mui/icons-material/ArrowRightRounded';

import { getProjectByProjectId, addTaskToProject, markTaskAsComplete } from '../../utils/data/projects';

import '../../styles/photoreview.css';

function PhotoReview() {
	const { projectid } = useParams();
	const [projectData, setProjectData] = useState({});
	const [photoIdx, setPhotoIdx] = useState(0);
	const [photoPage, setPhotoPage] = useState(1);
	const [photoPages, setPhotoPages] = useState(1);

	const [dateInput, setDateInput] = useState(new Date());
	const [statusInput, setStatusInput] = useState('approve');
	const [notesInput, setNotesInput] = useState('');

	const MAX_PHOTOS = 4;

	useEffect(() => {
		const fetchProject = async () => {
			const project = await getProjectByProjectId(projectid);
			setProjectData(project);
			setPhotoPages(Math.ceil(project.imageUrls.length / MAX_PHOTOS));
		};
		fetchProject();
	}, [projectid]);

	const handleViewPhotoButton = (idx) => {
		setPhotoIdx((photoPage - 1) * MAX_PHOTOS + idx);
	};


	const handleSubmitButton = async () => {
		// Approving adds the 'customer confirmation' task to the project, and marks the 'site review' task as complete.
		if (statusInput === 'approve') {
			await addTaskToProject(
				projectid, 
				'customer confirmation',
				`Approved: Estimated completion date is ${dateInput}`,
				true
			);
		}
		// Rejecting marks the 'site review' task as completed, and notes that it was rejected.
		if(statusInput === 'reject') {
			await markTaskAsComplete(projectid, 'site review', `Rejected: ${notesInput}`)
		}
	};

	if (Object.keys(projectData).length === 0) {
		return <div>Loading...</div>;
	} else if (projectData.imageUrls.length === 0) {
		return <div>No photos found for this project</div>;
	}

	return (
		<Box
			data-testid="photo-review-box"
			sx={{ display: 'flex', flexDirection: 'column', margin: 'auto' }}
		>
			<Typography variant="h1">Inspection Photo Review</Typography>
			<Typography>Project ID: {projectid}</Typography>
			<Typography>Address: {projectData.address}</Typography>
			<Card sx={{ display: 'flex', flexDirection: 'column', margin: 'auto' }}>
				<Typography>
					Photo {photoIdx + 1} of {projectData.imageUrls.length}
				</Typography>
				<img
					data-testid="photo-review-large"
					className="review-photo-large"
					src={projectData.imageUrls[photoIdx]}
				></img>
			</Card>
			<Box sx={{ display: 'flex', flexDirection: 'row', margin: 'auto' }}>
				{photoPage > 1 && (
					<IconButton
						data-testid="photo-review-previous-page-button"
						size="large"
						onClick={() => {
							setPhotoPage(photoPage - 1);
						}}
					>
						<ArrowLeftRoundedIcon fontSize="large" />
					</IconButton>
				)}
				{projectData.imageUrls
					.slice((photoPage - 1) * MAX_PHOTOS, photoPage * MAX_PHOTOS)
					.map((url, idx) => {
						return (
							<Card
								data-testid={`photo-review-thumbnail-${
									(photoPage - 1) * MAX_PHOTOS + idx
								}`}
								key={idx}
								sx={{
									m: 2,
									display: 'flex',
									flexDirection: 'column',
									border: 'grey 1px solid',
								}}
							>
								<img className="review-photo-thumbnail" src={url}></img>
								<Button
									data-testid={`photo-review-view-button-${
										(photoPage - 1) * MAX_PHOTOS + idx
									}`}
									onClick={(event) => handleViewPhotoButton(event, idx)}
									variant="contained"
									sx={{ maxWidth: '50%', margin: 'auto' }}
								>
									View
								</Button>
							</Card>
						);
					})}
				{photoPage < photoPages && (
					<IconButton
						data-testid="photo-review-next-page-button"
						size="large"
						onClick={() => {
							setPhotoPage(photoPage + 1);
						}}
					>
						<ArrowRightRoundedIcon fontSize="large" />
					</IconButton>
				)}
			</Box>
			<Box sx={{ display: 'flex', flexDirection: 'column', margin: 'auto' }}>
				<FormControl>
					<Select
						value={statusInput}
						size="small"
						onChange={(e) => {
							setStatusInput(e.target.value);
						}}
					>
						<MenuItem value="approve">Approve</MenuItem>
						<MenuItem value="reject">Reject</MenuItem>
					</Select>
					<br />
					{statusInput === 'reject' ? (
						<TextField
							label="Reason for Rejection"
							multiline
							variant="filled"
							type="text"
							margin="dense"
							required
							InputLabelProps={{ shrink: true }}
						/>
					) : (
						statusInput === 'approve' && (
							<TextField
								onChange={(e) => {
									setDateInput(e.target.value);
								}}
								label="Estimated Completion Date"
								variant="filled"
								type="date"
								margin="dense"
								required
								InputLabelProps={{ shrink: true }}
							/>
						)
					)}
					{statusInput !== '' && <Button onClick={handleSubmitButton} variant="contained">Submit</Button>}
				</FormControl>
			</Box>
		</Box>
	);
}

export default PhotoReview;
