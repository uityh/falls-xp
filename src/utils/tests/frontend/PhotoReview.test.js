import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { render, screen, cleanup } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { act } from 'react-dom/test-utils';

import PhotoReview from 'pages/PhotoReview';
import { getProjectByProjectId } from 'utils/data/projects';

afterEach(cleanup);

describe('PhotoReview', () => {
	it('renders loading div', () => {
		const tree = renderer.create(<PhotoReview />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});

jest.mock('utils/data/projects');

const mockProject = {
	id: 'F9ieqoKKXEmLOik946Nb',
	address: '308 Negra Arroyo Lane',
	assignedWorkers: [],
	cost: 0,
	customerId: 'eAE62yKcuRK4OCQYvmHc',
	salesRepId: 'smwmzTSYJzeGMbUC5HZm',
	status: 'not started',
	startDate: 'March 25, 2023 at 12:00:00 AM UTC-4',
	customerNotes: 'testing for images',
	tasks: 'initial inspection',
	imageUrls: [
		'https://firebasestorage.googleapis.com/v0/b/cs555-falls-xp.appspot.com/o/test_images%2Fkrabs.jpg?alt=media&token=56f1311f-4853-4e5e-9029-2fb4b1ab6a89',
		'https://firebasestorage.googleapis.com/v0/b/cs555-falls-xp.appspot.com/o/test_images%2Fsponge.png?alt=media&token=33eb6ca0-ea41-4242-98a7-98ce1d7cc837',
		'https://firebasestorage.googleapis.com/v0/b/cs555-falls-xp.appspot.com/o/test_images%2Fkrabs.jpg?alt=media&token=56f1311f-4853-4e5e-9029-2fb4b1ab6a89',
		'https://firebasestorage.googleapis.com/v0/b/cs555-falls-xp.appspot.com/o/test_images%2Fsponge.png?alt=media&token=33eb6ca0-ea41-4242-98a7-98ce1d7cc837',
		'https://firebasestorage.googleapis.com/v0/b/cs555-falls-xp.appspot.com/o/test_images%2Fkrabs.jpg?alt=media&token=56f1311f-4853-4e5e-9029-2fb4b1ab6a89',
		'https://firebasestorage.googleapis.com/v0/b/cs555-falls-xp.appspot.com/o/test_images%2Fsponge.png?alt=media&token=33eb6ca0-ea41-4242-98a7-98ce1d7cc837',
	],
};

describe('PhotoReview component', () => {
	beforeEach(() => {
		getProjectByProjectId.mockResolvedValue(mockProject);
	});

	afterEach(() => {
		jest.resetAllMocks();
	});

	it('Renders with the correct data', async () => {
		const { container } = render(<PhotoReview />);
		await act(async () => {
			expect(getProjectByProjectId).toHaveBeenCalledTimes(1);
		});
		expect(screen.getByTestId('photo-review-box')).toBeInTheDocument();
		expect(screen.getByTestId('photo-review-thumbnail-0')).toBeInTheDocument();
		expect(screen.getByTestId('photo-review-thumbnail-1')).toBeInTheDocument();
		expect(screen.getByTestId('photo-review-thumbnail-2')).toBeInTheDocument();
		expect(screen.getByTestId('photo-review-thumbnail-3')).toBeInTheDocument();


	});

	it('Renders the next page of images when the right arrow is clicked', async () => {
		const { container } = render(<PhotoReview />);
		await act(async () => {
			expect(getProjectByProjectId).toHaveBeenCalledTimes(1);
		});

		await act(async () => {
			userEvent.click(screen.getByTestId('photo-review-next-page-button'));
		});
		expect(screen.getByTestId('photo-review-thumbnail-4')).toBeInTheDocument();
		expect(screen.getByTestId('photo-review-thumbnail-5')).toBeInTheDocument();


	});


});

// describe('PhotoReview', () => {

// 	it('Page renders correctly with data',  async () => {

// 		await act( async () => {
// 			render(
// 				<MemoryRouter initialEntries={[{pathName: '/photo-review/F9ieqoKKXEmLOik946Nb'}]}>
// 					<PhotoReview />
// 				</MemoryRouter>
// 			);

// 		});

// 		await waitFor(() => {
// 			expect(screen.findByTestId('photo-review-box')).toBeInTheDocument();
// 			expect(screen.findByTestId('photo-review-thumbnail-0')).toBeInTheDocument();
// 			}
// 		);

// 	})
// });
