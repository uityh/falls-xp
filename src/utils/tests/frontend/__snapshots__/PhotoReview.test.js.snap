// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoReview renders loading div 1`] = `
<div>
  Loading...
</div>
`;
